version.go

```diff
@@ -129,73 +131,70 @@ func (v *Version) Metadata() string {
 	return v.metadata
 }
 
-// Increment version number,
-// How can be one of: patch, minor, major, prerelease
-func (v *Version) Inc(how string) bool {
-	if how == "patch" {
-		return v.IncPatch()
-	} else if how == "minor" {
-		return v.IncMinor()
-	} else if how == "major" {
-		return v.IncMajor()
-	}
-	return false
-}
-
-// Increment version number by the minor number.
+// IncPatch produces a new version instance with an incremented patch number.
+// Unsets metadata.
 // Unsets prerelease status.
-// Add +1 to patch number.
-func (v *Version) IncPatch() bool {
-	v.pre = ""
-	v.metadata = ""
-	v.patch += 1
-	return true
+func (v Version) IncPatch() Version {
+  vNext := v
+  vNext.metadata = ""
+  vNext.pre = ""
+	vNext.patch = v.patch+1
+	vNext.original = vNext.String()
+	return vNext
 }
 
-// Increment version number by the minor number.
+// IncMinor produces a new version instance with an incremented minor number.
+// Sets patch to 0.
+// Unsets metadata.
 // Unsets prerelease status.
-// Sets patch number to 0.
-// Add +1 to minor number.
-func (v *Version) IncMinor() bool {
-	v.pre = ""
-	v.metadata = ""
-	v.patch = 0
-	v.minor += 1
-	return true
+func (v Version) IncMinor() Version {
+  vNext := v
+  vNext.metadata = ""
+  vNext.pre = ""
+	vNext.patch = 0
+	vNext.minor = v.minor+1
+	vNext.original = vNext.String()
```

Comment:
Probably not safe to to try to preserve the original string, as it now describes the old value, not the current one